eng/pipelines/templates/steps/run-and-validate-linting.yml

```diff
@@ -0,0 +1,85 @@
+parameters:
+  - name: JavaBuildVersion
+    type: string
+    default: $(JavaBuildVersion)
+  - name: AdditionalLintingOptions
+    type: string
+    default: ''
+  - name: BuildParallelization
+    type: string
+    default: '2C'
+  - name: RunLinting
+    type: boolean
+    default: false
+  - name: ContinueOnError
+    type: boolean
+    default: false
+
+steps:
+  # maven dependency:tree needs to be able to resolve dependencies, so these should be installed.
+  - task: Maven@3
+    displayName: 'Build and Install'
+    inputs:
+      mavenPomFile: pom.xml
+      goals: 'install'
+      options: '$(DefaultOptions) -T ${{parameters.BuildParallelization}} -DskipTests $(DefaultSkipOptions) -pl $(ProjectList) -am'
+      mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
+      javaHomeOption: 'JDKVersion'
+      jdkVersionOption: ${{ parameters.JavaBuildVersion }}
+      jdkArchitectureOption: 'x64'
+      publishJUnitResults: false
+    continueOnError: ${{ parameters.ContinueOnError }}
+    condition: or(eq(coalesce(variables['IsLatestNonLtsJdk'], true), true), and(${{ parameters.RunLinting }}, succeeded()))
+
+  - task: Maven@3
+    displayName: 'Run SpotBugs, Checkstyle, and RevApi'
+    inputs:
+      mavenPomFile: pom.xml
+      options: '$(DefaultOptions) --no-transfer-progress -T 1C -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl $(ProjectList)'
+      mavenOptions: '$(MemoryOptions)'
+      javaHomeOption: 'JDKVersion'
+      jdkVersionOption: ${{ parameters.JavaBuildVersion }}
+      jdkArchitectureOption: 'x64'
+      publishJUnitResults: false
+      goals: 'checkstyle:check spotbugs:check revapi:check'
```

Comment:
Changing this goal from `verify` [causes an issue](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2936370&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=ba3cf43a-b2bf-50f9-f95f-93049ca44ad1) with libraries that don't use the azure-sdk-client-parent pom.